haystack/core/pipeline/base.py

```diff
@@ -0,0 +1,721 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
```

Comment:
This should contain the old `Pipeline` minus the `run` method. I had to change the type annotations of the class methods, something you might want to double check